lib/at_client/at_client.py

```diff
@@ -100,33 +100,42 @@ def put_public(self, keyName: str, value: str, ttr: int = 200) -> str:
 
         return: response from the secondary
         """
+        del ttr # TODO: implement ttr
+        
         from lib.at_client.at_utils import format_atSign
-        verb = 'update:public:%s%s %s' %(keyName, format_atSign(self.atSign), value)
-        del format_atSign
-        import time
-        time.sleep(1)
+        fullKeyName = keyName
+        if not namespace == None:
+            fullKeyName = keyName + '.' + namespace
+        verb = 'update:public:%s%s %s' %(fullKeyName, format_atSign(self.atSign), value)
+        del format_atSign, fullKeyName
+        from utime import sleep
+        sleep(1)
         print('sending verb: %s' % verb)
         response, command = self.send_verb(verb)
-        time.sleep(1)
-        del command, time
+        sleep(1)
+        del command, sleep
         response = response.replace('data:', '')
         return response
 
-    def get_public(self, keyName: str, otherAtSign:str) -> str:
+    def get_public(self, keyName: str, otherAtSign:str, namespace=None) -> str:
         """
         """
         from lib.at_client.at_utils import format_atSign
         otherAtSign = format_atSign(otherAtSign)
         del format_atSign
-        import time
+        from utime import sleep
 
         # get the public key
-        verb = 'plookup:bypassCache:true:%s%s' %(keyName, otherAtSign)
-        time.sleep(1)
+        fullKeyName = keyName
+        if not namespace == None:
+            fullKeyName = keyName + '.' + namespace
+        verb = 'plookup:bypassCache:true:%s%s' %(fullKeyName, otherAtSign)
```

Comment:
@TylerTrott is this namespace implementation fine?

Reply:
yeah, it's how I did it 👍 Nice work